src/Symfony/Component/Console/Helper/ProgressBar.php

```diff
@@ -243,6 +243,28 @@ public function setRedrawFrequency(int $freq)
         $this->redrawFreq = max($freq, 1);
     }
 
+    /**
+     * Returns an iterator that will automatically update the progress bar when iterated.
+     *
+     * If $iterator is countable, the progress bar max steps will be set to its count.
+     */
+    public function iterate(iterable $iterable): iterable
```

Comment:
would it make sense to pass the `$count` as optional arg in case $iterable is not countable?

Reply:
Not sure, in case `$iterable` is not countable you can do a `$this->setMaxSteps()` before the loop.

Having it as an argument could allow us to reset the max to `0` if `$iterable` is not countable. That could be a good thing in case the bar is first used with a count and then without it.